lib/filter-stream.js

```diff
@@ -168,39 +223,52 @@ function FilterStream( blockMap, options ) {
    */
   _transform( chunk, _, next ) {
 
-    this._buffers.push( chunk )
-    this._bytes += chunk.length
+    debug( 'position', this.position )
+    debug( 'chunk', chunk.length, chunk )
 
-    if( this._bytes < this.blockSize )
-      return next()
+    this.bytesRead += chunk.length
+    this.blocksRead += chunk.length / this.blockSize
 
-    var buffer = Buffer.concat( this._buffers )
-    var offset = 0
+    if( this.currentRange == null ) {
+      debug( 'no current range' )
+      return next()
+    }
 
-    while( buffer.length - offset >= this.blockSize ) {
-      this._pushBlock( buffer.slice( offset, offset + this.blockSize ) )
-      offset = offset + this.blockSize
+    if( !this._rangeInChunk( chunk ) ) {
+      debug( 'chunk:ignore' )
```

Comment:
I'm not sure I get the purpose of FilterStream.
As I understand it, it strips out the parts of the input stream that are not in the blockmap. What would be the use case for this?